crypto/dsa/dsa_ossl.c

```diff
@@ -393,3 +394,30 @@ static int dsa_finish(DSA *dsa)
     BN_MONT_CTX_free(dsa->method_mont_p);
     return 1;
 }
+
+/*
+ * Compute the inverse of k modulo q.
+ * Since q is prime, Fermat's Little Theorem applies, which reduces this to
+ * mod-exp operation.  Both the exponent and modulus are public information
+ * so a constant time mod-exp shouldn't be required.  A newly allocated BIGNUM
+ * is returned which the caller must free.
+ */
+static BIGNUM *dsa_inverse_mod(const BIGNUM *k, const BIGNUM *q, BN_CTX *ctx)
```

Comment:
Also, wouldn't a name like `dsa_mod_inverse_fermat` be more expressive than just `dsa_inverse_mod`?

Reply:
Yes, I'll change this.